subsys/event_manager/event_manager.c

```diff
@@ -40,24 +60,11 @@ static void log_event(const struct event_header *eh)
 	}
 
 	if (et->log_event) {
-		char log_buf[CONFIG_EVENT_MANAGER_EVENT_LOG_BUF_LEN];
-
-		int pos = et->log_event(eh, log_buf, sizeof(log_buf));
-
-		if (pos < 0) {
-			log_buf[0] = '\0';
-		} else if (pos >= sizeof(log_buf)) {
-			BUILD_ASSERT(sizeof(log_buf) >= 2,
-					 "Buffer invalid");
-			log_buf[sizeof(log_buf) - 2] = '~';
-		}
-
-		if (IS_ENABLED(CONFIG_EVENT_MANAGER_LOG_EVENT_TYPE)) {
-			LOG_INF("e: %s %s", et->name, log_strdup(log_buf));
+		if (CONFIG_EVENT_MANAGER_EVENT_LOG_BUF_LEN > 1) {
+			log_event_using_buffer(eh, et);
 		} else {
-			LOG_INF("%s", log_strdup(log_buf));
+			(void)et->log_event(eh, NULL, 0);
```

Comment:
Maybe assert that it's not one here? It may get quite counterintuitive here.

Reply:
It turns out there is a buffer, but for some reason it's not used at all.

Reply:
Do not understand you quite well here. If CONFIG_EVENT_MANAGER_EVENT_LOG_BUF_LEN is equal 0 or 1 we wont use any buffer I can add comment that passing NULL for buffer is done due to legacy code.

Reply:
When I define buffer of size one in Kconfig, it's not used at all. That seemed a bit counterintuitive for me, that's why I proposed adding e.g. build assert to explicitly inform that size of 1 is invalid.

It's understandable that when I define buffer of zero size, NULL is passed instead (for me it's intuitive since I still cannot use the buffer at all). The same behavior is no longer intuitive if size of user defined buffer is 1. That's why I proposed introducing build assert instead of not using the buffer of size 1.